### PR TITLE
fix(#4365): safely handle missing brightness fields in LYP files

### LINKED ISSUE #4365: lyp_to_dataclass &  AttributeError: 'NoneType' object has no attribute 'text'
**Describe the bug**
What's the bug?
AttributeError                            Traceback (most recent call last)
Cell In[3], line 1
----> 1 lyp_to_dataclass("layer.lyp", map_name="Layer")

File [~\anaconda3\envs\standard\Lib\site-packages\gdsfactory\technology\layer_map.py:43](http://localhost:8888/lab/tree/Box/Simu/~/anaconda3/envs/standard/Lib/site-packages/gdsfactory/technology/layer_map.py#line=42), in lyp_to_dataclass(lyp_filepath, overwrite, sort_by_name, map_name, output_filepath)
     33         raise ValueError(
     34             f"Argument 'map_name' must be a valid python identifier, but {map_name} is not."
     35         )
     37     script = f"""from gdsfactory.typings import Layer
     38 from gdsfactory.technology.layer_map import LayerMap
     39 
     40 
     41 class {map_name}(LayerMap):
     42 """
---> 43     lys = LayerViews.from_lyp(filepathin)
     44     maybe_sort = sorted if sort_by_name else lambda seq: seq
     45     for layer_name, layer in maybe_sort(lys.get_layer_views().items()):

File [~\anaconda3\envs\standard\Lib\site-packages\gdsfactory\technology\layer_views.py:1161](http://localhost:8888/lab/tree/Box/Simu/~/anaconda3/envs/standard/Lib/site-packages/gdsfactory/technology/layer_views.py#line=1160), in LayerViews.from_lyp(filepath, layer_pattern)
   1159 layer_views = {}
   1160 for properties_element in root.iter("properties"):
-> 1161     lv = LayerView.from_xml_element(
   1162         properties_element, layer_pattern=layer_pattern
   1163     )
   1164     if lv:
   1165         hp = lv.hatch_pattern

File [~\anaconda3\envs\standard\Lib\site-packages\gdsfactory\technology\layer_views.py:758](http://localhost:8888/lab/tree/Box/Simu/~/anaconda3/envs/standard/Lib/site-packages/gdsfactory/technology/layer_views.py#line=757), in LayerView.from_xml_element(cls, element, layer_pattern)
    746 if (
    747     line_style is not None
    748     and line_style.text is not None
    749     and re.match(r"I\d+", line_style.text)
    750 ):
    751     line_style = list(_klayout_line_styles.keys())[int(line_style.text[1:])]  # type: ignore[assignment]
    753 lv = LayerView(
    754     name=name,
    755     layer=cls._process_layer(element.find("source").text, layer_pattern),  # type: ignore[union-attr,arg-type]
    756     fill_color=getattr(element.find("fill-color"), "text", None),
    757     frame_color=getattr(element.find("frame-color"), "text", None),
--> 758     fill_brightness=element.find("fill-brightness").text or 0,  # type: ignore[union-attr]
    759     frame_brightness=element.find("frame-brightness").text or 0,  # type: ignore[union-attr]
    760     hatch_pattern=hatch_pattern or None,
    761     line_style=line_style
    762     if line_style is not None and len(line_style) > 0
    763     else None,
    764     valid=getattr(element.find("valid"), "text", True),
    765     visible=getattr(element.find("visible"), "text", True),
    766     transparent=getattr(element.find("transparent"), "text", False),
    767     width=getattr(element.find("width"), "text", None),
    768     marked=getattr(element.find("marked"), "text", False),
    769     xfill=getattr(element.find("xfill"), "text", False),
    770     animation=getattr(element.find("animation"), "text", False),
    771     layer_in_name=layer_in_name,
    772 )
    774 # Add only if needed, so we can filter by defaults when dumping to yaml
    775 group_members: builtins.dict[str, LayerView] = {}

AttributeError: 'NoneType' object has no attribute 'text'
**To Reproduce**
What's the code to reproduce the behavior? What commands or code did you write to get the error?
Make sure you include the all code needed for others to reproduce your issue.
from gdsfactory.technology import LayerLevel, LayerMap, LayerStack, LayerViews, lyp_to_dataclass
lyp_to_dataclass("layer.lyp")
**Expected behavior**
What would you like to happen?
no attribute check
**Suggested fix**
How could we fix the bug? As an open-source project, we welcome your suggestions and PRs.
 Use "getattr()" to examine the existence of an attribute. 

**Environment (please complete the following information):**

- [y] I have reviewed the documentation, discussions and issues and found no relevant solution.
- [y] I understand that this is an open-source project, and maintainers may not have the resources to address every issue. I am prepared to contribute by fixing the issue myself or hiring someone to do so, if needed, and accept that the issue may not be resolved otherwise.
- [y] I am using the latest version of GDSFactory, with Python 3.11, 3.12, or 3.13.

Please provide the output for the following code:

```python
import sys
print(sys.version)
print(sys.executable)

import gdsfactory as gf
gf.config.print_version_plugins()
```
                                                     Modules                                                      
┏━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Package ┃ version                                       ┃                                           Path ┃
┡━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│         python │ 3.11.13 | packaged by conda-forge | (main,    │ C:\Users\anaconda3\envs\standard\py… │
│                │ Jun  4 2025, 14:39:58) [MSC v.1943 64 bit     │                                                │
│                │ (AMD64)]                                      │                                                │
│     gdsfactory │ 9.34.0                                        │ C:\Users\anaconda3\envs\standard\Li… │
│         devsim │ not installed                                 │                                                │
│        femwell │ not installed                                 │                                                │
│ gdsfactoryplus │ not installed                                 │                                                │
│       gplugins │ not installed                                 │                                                │
│       kfactory │ 2.3.0                                         │ ['C:\\Users\\anaconda3\\envs\\stan… │
│         lumapi │ not installed                                 │                                                │
│           meep │ not installed                                 │                                                │
│           meow │ not installed                                 │                                                │
│            ray │ not installed                                 │                                                │
│            sax │ not installed                                 │                                                │
│         tidy3d │ not installed                                 │                                                │
│          vlsir │ not installed                                 │                                                │
└────────────────┴───────────────────────────────────────────────┴────────────────────────────────────────────────┘

### PR DESCRIPTION
**Pull request by dhineshkumar2512v**
## Summary
Replaced fragile `.find().text` calls with safer `getattr` checks in `layer_views.py` to prevent crashes when LYP files lack brightness metadata.

Fixes #4365

## Summary by Sourcery

Bug Fixes:
- Prevent crashes when LYP files omit fill or frame brightness metadata by defaulting missing values to zero.

### CODE REVIEW

File: gdsfactory/technology/layer_views.py [hunk 1]
```diff
@@ -755,8 +755,9 @@ def from_xml_element(
             layer=cls._process_layer(element.find("source").text, layer_pattern),  # type: ignore[union-attr,arg-type]
             fill_color=getattr(element.find("fill-color"), "text", None),
             frame_color=getattr(element.find("frame-color"), "text", None),
-            fill_brightness=element.find("fill-brightness").text or 0,  # type: ignore[union-attr]
-            frame_brightness=element.find("frame-brightness").text or 0,  # type: ignore[union-attr]
+            fill_brightness=getattr(element.find("fill-brightness"), "text", None) or 0,
+            frame_brightness=getattr(element.find("frame-brightness"), "text", None)
+            or 0,
```
Comment (sebastian-goeldi) [inline gdsfactory/technology/layer_views.py]:
```suggestion
            fill_brightness=getattr(element.find("fill-brightness"), "text", 0),
            frame_brightness=getattr(element.find("frame-brightness"), "text", 0),
```

No need for testing for None if we want it to be 0 anyways in that case ;)

**Review by sebastian-goeldi — CHANGES_REQUESTED** [review]

**Review by joamatab — APPROVED** [review]
